### PR TITLE
Add support for publicKey from BEP46

### PR DESCRIPTION
**Pull request by RangerMauve**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[x] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**

- Added encode/decode rules for `xs=urn:btpk:PUBLIC_KEY` to parse out a `publicKey` based on [BEP 46](http://bittorrent.org/beps/bep_0046.html)
- Encoding got modified to only use URI encode for `xs` strings that don't start with `urn:btpk:`

**Which issue (if any) does this pull request address?**

This will make life easier for me in [mutable-webtorrent](https://github.com/RangerMauve/mutable-webtorrent)

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

**Comment by RangerMauve** [comment]
@alxhotel Is there anything else I can do to get this merged in? 😅

File: index.js [hunk 1]
```diff
@@ -74,7 +74,18 @@ function magnetURIDecode (uri) {
       }
     })
   }
+
+  if (result.xs) {
+    const xss = Array.isArray(result.xs) ? result.xs : [result.xs]
+    xss.forEach(xs => {
+      if ((m = xs.match(/^urn:btpk:(.{64})/))) {
+        result.publicKey = m[1]
```
Comment (feross) [inline index.js:82]: This is a hex string, right? Why no `.toLowerCase()` like we did for `xt` above?

File: index.js [hunk 2]
```diff
@@ -130,7 +143,11 @@ function magnetURIEncode (obj) {
         if ((i > 0 || j > 0) && ((key !== 'kt' && key !== 'so') || j === 0)) result += '&'
 
         if (key === 'dn') val = encodeURIComponent(val).replace(/%20/g, '+')
-        if (key === 'tr' || key === 'xs' || key === 'as' || key === 'ws') {
+        if (key === 'tr' || key === 'as' || key === 'ws') {
+          val = encodeURIComponent(val)
+        }
+        // Don't URI encode BEP46 keys
+        if (key === 'xs' && !val.startsWith('urn:btpk')) {
```
Comment (feross) [inline index.js:150]: Can you change this to `startsWith('urn:btpk:')` (note the trailing colon!)

**Review by feross — CHANGES_REQUESTED** [review]
LGTM! This is basically ready to merge, except two nits I mentioned inline.

**Comment by feross** [comment]
@RangerMauve Yep, see the inline comments! @alxhotel Thanks for updating this PR, and bringing attention back to it

**Comment by RangerMauve** [comment]
@feross Thank you for the review! Just added the changes you mentioned. 😁

**Review by feross — APPROVED** [review]
LGTM

**Comment by feross** [comment]
Released as 6.1.0

**Comment by RangerMauve** [comment]
Sweet. Thank you for merging this in! I'm excited to get the public key stuff working in higher levels of the stack, too. 😁

**Pull request merged**